### PR TITLE
Fix `InternalAffairs/ExampleDescription` offenses on master

### PR DESCRIPTION
**Pull request by tejasbubane**
CI is failing with latest rubocop https://app.circleci.com/pipelines/github/rubocop/rubocop-rspec/887/workflows/cb3ac5d9-a753-42fd-a26d-86b9e5bd8c69/jobs/20828

Rubocop change: https://github.com/rubocop/rubocop/pull/9797 - Released in rubocop `v1.15.0`

---

Before submitting the PR make sure the following are checked:

* [x] Feature branch is up-to-date with `master` (if not - rebase it).
* [x] Squashed related commits together.
* [ ] Added tests.
* [ ] Updated documentation.
* [ ] Added an entry to the `CHANGELOG.md` if the new code introduces user-observable changes.
* [x] The build (`bundle exec rake`) passes (be sure to run this locally, since it may produce updated documentation that you will need to commit).

If you have created a new cop:

* [ ] Added the new cop to `config/default.yml`.
* [ ] The cop is configured as `Enabled: pending` in `config/default.yml`.
* [ ] The cop documents examples of good and bad code.
* [ ] The tests assert both that bad code is reported and that good code is not reported.
* [ ] Set `VersionAdded` in `default/config.yml` to the next minor version.

If you have modified an existing cop's configuration options:

* [ ] Set `VersionChanged` in `config/default.yml` to the next major version.

### CODE REVIEW

File: spec/rubocop/cop/rspec/empty_line_after_final_let_spec.rb [hunk 1]
```diff
@@ -65,30 +65,32 @@
...
 
-      it { expect(a).to eq(b) }
-    end
+        it { expect(a).to eq(b) }
+      end
     RUBY
   end
 
-  it 'flags missing empty line after the comment that comes after last let' do
+  it 'register an offense for missing empty line after the comment '\
```
Comment (koic) [inline spec/rubocop/cop/rspec/empty_line_after_final_let_spec.rb:92]:
```suggestion
  it 'registers an offense for missing empty line after the comment '\
```

Comment (tejasbubane) [reply spec/rubocop/cop/rspec/empty_line_after_final_let_spec.rb]: Fixed.

File: spec/rubocop/cop/rspec/empty_line_after_final_let_spec.rb [hunk 2]
```diff
@@ -134,7 +137,7 @@
     RUBY
   end
 
-  it 'ignores empty lines between the lets' do
```
Comment (pirj) [inline spec/rubocop/cop/rspec/empty_line_after_final_let_spec.rb:137]: The point of this example is different. I believe the best way is to add a newline between `let` and `it` and use "does not register an offence for empty lines between the lets".

File: spec/rubocop/cop/rspec/file_path_spec.rb [hunk 3]
```diff
@@ -71,119 +71,119 @@
...
 
-  it 'ignores shared examples' do
+  it 'does not register an offense for shared examples' do
     expect_no_offenses(<<-RUBY, 'spec/models/user.rb')
       shared_examples_for 'foo' do; end
     RUBY
   end
 
-  it 'skips specs that do not describe a class / method' do
+  it 'does not register an offense for describe a class / method' do
```
Comment (pirj) [inline spec/rubocop/cop/rspec/file_path_spec.rb:80]: It is more like "does not register an offence for example groups do not describe a class / method".

File: spec/rubocop/cop/rspec/file_path_spec.rb [hunk 4]
```diff
@@ -71,119 +71,119 @@
...
-  it 'allows flexibility with predicates' do
+  it 'does not register an offense for flexibility with predicates' do
     filename = 'some/class/thing_predicate_spec.rb'
     expect_no_offenses(<<-RUBY, filename)
       describe Some::Class, '#thing?' do; end
     RUBY
   end
 
-  it 'allows flexibility with operators' do
+  it 'does not register an offense for flexibility with operators' do
```
Comment (pirj) [inline spec/rubocop/cop/rspec/file_path_spec.rb:186]: Those two do not read right.
In any case, it allows flexibility not only for operators/predicates:
```
            name_pattern(method_name),
            '[^/]*_spec\.rb'
```
but actually for any spec:
```ruby
  it 'allows flexibility with predicates' do
    filename = 'some/class/thing_predicate_spec.rb'
    expect_no_offenses(<<-RUBY, filename)
      describe Some::Class, '#thing' do; end
    RUBY
  end
```
passes just fine.

I'd suggest changing the first example to "does not register an offence for an arbitrary spec suffix".

And the second to "does not register an offence for an arbitrary spec name for an operator method".

File: spec/rubocop/cop/rspec/overwriting_setup_spec.rb [hunk 5]
```diff
@@ -55,7 +55,7 @@
     RUBY
   end
 
-  it 'handles unnamed subjects' do
+  it 'registers an offense for unnamed subjects' do
```
Comment (pirj) [inline spec/rubocop/cop/rspec/overwriting_setup_spec.rb:58]: This is not correct. I believe it's more like "registers an offense for overriding an unnamed subject".

File: spec/rubocop/cop/rspec/return_from_stub_spec.rb [hunk 6]
```diff
@@ -176,15 +176,15 @@
       RUBY
     end
 
-    it 'ignores stubs without return value' do
+    it 'does not register an offense for stub without return value' do
```
Comment (pirj) [inline spec/rubocop/cop/rspec/return_from_stub_spec.rb:179]: Nitpick.
'does not register an offense for A stub without return value' or
'does not register an offense for stubs without return values'.

**Review by pirj — APPROVED** [review]
Thank you so much!
Diversity in example descriptions was triggering my OCD for so long.

**Comment by tejasbubane** [comment]
@pirj Fixed as per your review comments.

File: spec/rubocop/cop/rspec/file_path_spec.rb [hunk 7]
```diff
@@ -71,119 +71,120 @@
...
-  it 'allows flexibility with predicates' do
+  it 'does not register an offence for an arbitrary spec suffix' do
     filename = 'some/class/thing_predicate_spec.rb'
     expect_no_offenses(<<-RUBY, filename)
       describe Some::Class, '#thing?' do; end
     RUBY
   end
 
-  it 'allows flexibility with operators' do
+  it 'does not register an offense for flexibility with operators' do
```
Comment (pirj) [inline spec/rubocop/cop/rspec/file_path_spec.rb:187]: I insist on "does not register an offence for an arbitrary spec name for an operator method".

Comment (tejasbubane) [reply spec/rubocop/cop/rspec/file_path_spec.rb]: Changed.

**Comment by bquorning** [comment]
Could you please add to the PR description a link to the change in rubocop/rubocop that prompted these changes?

**Comment by tejasbubane** [comment]
> Could you please add to the PR description a link to the change in rubocop/rubocop that prompted these changes?

@bquorning Added

**Comment by pirj** [comment]
Please ping us when your other PRs builds pass.
And thanks again!

**Pull request merged**